Add extra args to collate to handle edge cases

bethac07 opened the pull request:
# Description

Collate handles the standard cases pretty well at this point, but there are a number of edge cases I've been writing extra code in my own fork to handle; it would be nice to add first-class support

## What is the nature of your change?

- [x] Enhancement (adds functionality).
- [x] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

bethac07 committed:
Add extra args to collate to handle edge cases

bethac07 committed:
black

d33bs commented:
Hello, this is a courtesy notice that Pcytominer's `master` branch has been renamed to `main` as part of changes merged in #303. While changes here should be seamless, GitHub is unable to change your local git environment and offers the following guidance when it comes to this. Thank you for your understanding and please don't hesitate to reach out with any questions or concerns.

![image](https://github.com/cytomining/pycytominer/assets/3738008/b0006f70-f59e-464a-8f51-1b248a907f3e)

d33bs commented:
Hi @bethac07 - thanks for your efforts on this! I wanted to check in to see if you're still planning to move forward with this PR. If there's any way we may assist or if you have any questions just let us know. Thanks for taking a moment to make a contribution here!

bethac07 committed:
Merge branch 'main' into csv_cleanup

pre-commit-ci-lite[bot] committed:
[pre-commit.ci lite] apply automatic fixes

bethac07 committed:
Update docstrings

pre-commit-ci-lite[bot] committed:
[pre-commit.ci lite] apply automatic fixes

bethac07 commented:
Hi @d33bs,

Just for funsies, I've rebased it and cleaned up some docstrings, but if collate is being deprecated, spending a bunch of time writing tests (which is all this needed aside from those docstrings) is probably not worth my nor your time. In a perfect world, despite the deprecation and ruff comments I'd still want to merge it so that it can be accessed if needed in a pre 2.0 `main` commit, but if you don't want to do that we'll figure something else out (like move to a fork). Thanks!

d33bs commented:
Thanks @bethac07 ! I understand what you mean about this and feel similarly. Just the same, I'd like to keep our CI tests passing to help avoid confusion with future changes. Would it be alright with you for me to modify the code so it passes `ruff` checks (while retaining the existing operational results where possible)?

bethac07 commented:
Sure, fine with me :)

d33bs committed:
address linting errors; add docs on deprecation

d33bs commented:
Hi @bethac07 - I made some changes to help us pass linting checks. Double checking: is this ready for review and moving forward (the PR is still in draft state and includes `[WIP]`)?

bethac07 commented:
That was just the tests, if we can pull without, it's all set! Thanks much for the help

d33bs reviewed (commented):
Hi @bethac07 - nice job, thanks for the additions. I wanted to make sure this code was readied in context with #301 in mind and provided some feedback that I'd like to get your thoughts on before proceeding with a merge. Thanks in advance for your attention to these.

gwaybio commented:
closing for staleness - please feel free to reopen if changes are needed!
